reconcile/utils/vault.py

```diff
@@ -315,7 +315,16 @@ def _write_v1(self, path, data):
 
     def _list(self, path: str) -> dict:
         try:
-            return self._client.list(path)
+            response = self._client.list(path)
+            if response is not None:
+                return response
+            # If vault path is from a v2 engine hvac.list(path) returns None,
+            # to list secrets from a v2 engine we need to use hvac.secrets.kv.v2.list_secrets
+            secret_path = path.split("/", 1)
```

Comment:
suggestion: I think this will be a bit clearer:

```suggestion
            mount_point, secret_path = path.split("/", 1)
```

Reply:
👍